code/game/objects/structures/dropship_equipment.dm

```diff
@@ -1039,3 +1042,328 @@
 	deployed_table.layer = ABOVE_OBJ_LAYER + 0.01 //make sure its directly ABOVE the layer
 	deployed_table.loc = loc
 	icon_state = "table2-idle"
+
+//////////////// RAPPEL SYSTEM /////////////////
+
+//Rappel target selection action
+/datum/action/innate/rappel_designate
+	name = "Designate rappel point"
+	action_icon = 'icons/mecha/actions_mecha.dmi'
+	action_icon_state = "mech_zoom_on"
+	var/obj/structure/dropship_equipment/rappel_system/origin
+
+/datum/action/innate/rappel_designate/Activate()
+	var/mob/C = target
+	var/mob/camera/aiEye/remote/remote_eye = C.remote_control
+	var/turf/target_turf = get_turf(remote_eye)
+	var/area/target_area = get_area(target_turf)
```

Comment:
Why are you setting these vars if there is a possibility you do not use them.
Define vars latest as possible